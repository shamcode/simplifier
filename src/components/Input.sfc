<template>
    <input
        id={{id}}
        placeholder={{placeholder}}
        tabindex="{{tabindex}}"
        name={{name}}
    />
</template>

<script>
    function Input( options ) {
        options( {
            [ ref( 'placeholder' ) ]: '',
            [ ref( 'tabindex' ) ]: 1
        } );
    }

    export default Component( Template, Input );
</script>
