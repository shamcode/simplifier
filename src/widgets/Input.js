import { options } from 'sham-ui';
import Input from './Input.sht';

export default class extends Input {
    @options
    get placeholder() {
        return '';
    }

    @options
    get tabindex() {
        return 1;
    }
}