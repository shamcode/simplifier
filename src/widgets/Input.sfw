<template>
    <input
            id={{id}}
            placeholder={{placeholder}}
            tabindex="{{tabindex}}"
            name={{name}}
    />
</template>

<script>
    import { options } from 'sham-ui';

    export default class Input extends Template {
        @options placeholder = '';
        @options tabindex = 1;
    }
</script>