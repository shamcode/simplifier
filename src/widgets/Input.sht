<input
        id={{id}}
        placeholder={{placeholder}}
        tabindex="{{tabindex}}"
        name={{name}}
/>