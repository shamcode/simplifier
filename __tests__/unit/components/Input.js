import Input from '../../../src/components/Input.sfc';
import renderer from 'sham-ui-test-helpers';

it( 'renders correctly', () => {
    const meta = renderer( Input );

    expect( meta.component.container.innerHTML ).toEqual( '<input placeholder="" tabindex="1">' );
    expect( meta.toJSON() ).toMatchSnapshot();
} );

it( 'with props', () => {
    const meta = renderer( Input, {
        id: 'expression-input',
        placeholder: 'a && b',
        tabindex: 2,
        name: 'expression'
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );
