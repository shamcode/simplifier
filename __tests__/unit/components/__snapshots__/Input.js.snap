// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "Options": Object {
    "ID": "component",
    "placeholder": "",
    "tabindex": 1,
  },
  "html": "<input placeholder=\\"\\" tabindex=\\"1\\">",
}
`;

exports[`with props 1`] = `
Object {
  "Options": Object {
    "ID": "component",
    "id": "expression-input",
    "name": "expression",
    "placeholder": "a && b",
    "tabindex": 2,
  },
  "html": "<input id=\\"expression-input\\" placeholder=\\"a &amp;&amp; b\\" tabindex=\\"2\\" name=\\"expression\\">",
}
`;
